src/NHibernate/Util/ArrayHelper.cs

```diff
@@ -179,45 +179,12 @@ public static int CountTrue(bool[] array)
 
 		public static bool ArrayEquals<T>(T[] a, T[] b)
```

Comment:
Turns out this magic already exists in the .NET Fx:

`Array` implements `IStructuralEquatable` interface:

 var a = new[] {1};
            var b = new[] {1};
            IStructuralEquatable aa = a;

            var @equals = aa.Equals(b, EqualityComparer<int>.Default);

Reply:
But that magic works by comparing everything as objects as it uses the non generic `IEqualityComparer` interface, causing value types to be boxed.
The `ArrayEquals(byte[] a, byte[] b)` overload would be likely less efficient if re-implemented with this.
The way I have re-implemented this (with the last commit) also allows `ArrayEquals<T>` to avoid boxing.